### PR TITLE
CloudWatch: Run query on blur in logs query field

### LINKED ISSUE #47224: CloudWatch: Logs Query Editor should run query on blur
**What happened**:
When a logs query is edited it doesn't run unless the refresh button is hit for the panel, even if it's a valid query.

**What you expected to happen**:
When the Log Groups are changed or the text query is blurred it should call `onRunQuery`. Alternatively, a run query button could be added to the editor that calls `onRunQuery`.

**How to reproduce it (as minimally and precisely as possible)**:
Open a panel for a Cloudwatch log query and set values for Log Groups and the text query (the help button can autofill one)

**Anything else we need to know?**:

**Environment**:
- Grafana version:
- Data source type & version:
- OS Grafana is installed on:
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by fridgepoet**
**What this PR does / why we need it**:

#47224 noted that queries should be run whenever the `MultiSelect` or `QueryField` for `LogsQueryField` are blurred.

**Which issue(s) this PR fixes**:

Fixes #47224

**Special notes for your reviewer**:
- Shortcut to run just the cloudwatch tests: `yarn jest -i --testPathPattern=/cloudwatch/components/`
- Create a CloudWatch panel and specify at least one log group along with a query and note whenever either field is blurred the "refresh dashboard" icon spins to indicate the query is being run.

### CODE REVIEW

File: public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx [hunk 1]
```diff
@@ -344,7 +345,6 @@ export class CloudWatchLogsQueryField extends React.PureComponent<CloudWatchLogs
               additionalPlugins={this.plugins}
               query={query.expression ?? ''}
               onChange={this.onChangeQuery}
-              onBlur={this.props.onBlur}
```
Comment (iwysiu) [inline public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx:347]: Should the query field also run on blur? (Assuming that there's at least one selectedLogGroup)

File: public/app/plugins/datasource/cloudwatch/components/LogsQueryField.test.tsx [hunk 2]
```diff
@@ -6,12 +6,40 @@ import { DescribeLogGroupsRequest } from '../types';
...
+
+    render(
+      <CloudWatchLogsQueryField
+        absoluteRange={{ from: 1, to: 10 }}
+        exploreId={ExploreId.left}
+        datasource={
+          {
+            getVariables: () => [],
+            getRegions: () => Promise.resolve([]),
+          } as any
```
Comment (sarahzinger) [inline public/app/plugins/datasource/cloudwatch/components/LogsQueryField.test.tsx:28]: I believe we have some sort of function that's like setupMockDatasource or something like that, might be nice to use that right?

**Review by sarahzinger — APPROVED** [review]
Looks good to me, I have the same q as isabella, I think if we update either the log group or the query editor I would expect we'd run on query on blur I think.

File: public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx (hunk 1 above)
Comment (asimpson) [reply public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx:347]: @sarahzinger @iwysiu `QueryField` is configured by default to run the query on blur: https://github.com/grafana/grafana/blob/main/packages/grafana-ui/src/components/QueryField/QueryField.tsx#L189 Passing the prop would be needed if we wanted different behavior.

Comment (iwysiu) [reply public/app/plugins/datasource/cloudwatch/components/LogsQueryField.tsx:347]: 👍  thanks for clarifying

**Review by iwysiu — APPROVED** [review]

**Pull request merged**